Frameworks/Foundation/NSKVOSupport.mm

```diff
@@ -667,16 +668,7 @@ - (void)willChange:(NSKeyValueChange)changeKind valuesAtIndexes:(NSIndexSet*)ind
 
             if ((options & NSKeyValueObservingOptionOld) && localChangeKind != NSKeyValueChangeInsertion) {
                 // For to-many mutations, we can't get the old values at indexes that have not yet been inserted.
-                change[NSKeyValueChangeOldKey] =
-                    _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keypath, keyObserver, indexes);
-            }
-
-            if (options & NSKeyValueObservingOptionPrior) {
-                id observer = keypathObserver.observer;
-                void* context = keypathObserver.context;
-                [change setObject:@(YES) forKey:NSKeyValueChangeNotificationIsPriorKey];
-                [observer observeValueForKeyPath:keypath ofObject:rootObject change:change context:context];
-                [change removeObjectForKey:NSKeyValueChangeNotificationIsPriorKey];
+                change[NSKeyValueChangeOldKey] = _valueForPendingChangeAtIndexes(notifyingObject, key, rootObject, keyObserver, indexes);
```

Comment:
Incidentally, this is why that function pulled `Indexes` out of the change dictionary -- it was the ideal way to communicate that we had degraded a collection mutation to a single setter mutation across the `will`->`did` divice.